components/hab/src/main.rs

```diff
@@ -304,10 +305,38 @@ fn sub_pkg_export(ui: &mut UI, m: &ArgMatches) -> Result<()> {
 }
 
 fn sub_pkg_hash(m: &ArgMatches) -> Result<()> {
-    let source = m.value_of("SOURCE").unwrap(); // Required via clap
-
     init();
-    command::pkg::hash::start(&source)
+    match m.value_of("SOURCE") {
+        Some(source) => {
+            // hash single file
+            command::pkg::hash::start(&source)
+        },
+        None => {
+            // read files from stdin
+            let stdin = io::stdin();
+            for line in stdin.lock().lines() {
+                match line {
+                    Ok(line) => {
+                        let file = line.trim_right();
+                        if Path::new(&file).is_file() {
+                            match command::pkg::hash::start(&file) {
+                                Ok(_) => (),
+                                Err(error) => {
+                                    writeln!(io::stderr(), "{}: error hashing file: {}", &file, error).unwrap();
+                                }
+                            }
+                        } else {
+                            writeln!(io::stderr(), "{}: No such file or directory", &file).unwrap();
+                        }
+                    },
+                    Err(error) => {
+                        writeln!(io::stderr(), "error: {}", error).unwrap();
+                    },
```

Comment:
ahh i see. I'd still return a `Result` with the `err` in it instead of writing to stderr from this function. This kiips the actual writing of the error messages centralized using the `ui` in `main::start`